netbox/netbox/models/__init__.py

```diff
@@ -58,6 +59,32 @@ class NetBoxModel(CloningMixin, NetBoxFeatureSet, models.Model):
     class Meta:
         abstract = True
 
+    def clean(self):
+        """
+        Validate the model for GenericForeignKey fields to ensure that the content type and object ID exist.
+        """
+        super().clean()
+
+        for field in self._meta.get_fields():
+            if isinstance(field, GenericForeignKey):
+                if getattr(self, field.ct_field) is None and getattr(self, field.fk_field) is not None:
+                    raise ValidationError({
+                        field.ct_field: "This field cannot be null.",
+                    })
+                if getattr(self, field.fk_field) is None and getattr(self, field.ct_field) is not None:
+                    raise ValidationError({
+                        field.fk_field: "This field cannot be null.",
+                    })
+
+                if getattr(self, field.ct_field) and getattr(self, field.fk_field):
+                    klass = getattr(self, field.ct_field).model_class()
+                    try:
+                        klass.objects.get(pk=getattr(self, field.fk_field))
```

Comment:
Calling `get()` retrieves and instantiates the entire object from the database. Since we only need to check for its existence, we can do:

```suggestion
                        klass.objects.filter(pk=getattr(self, field.fk_field)).exists()
```